File: back/pki/src/main/java/com/example/pki/model/Certificate.java
package com.example.pki.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Certificate {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(unique = true)
    private String serialNumber;
    private String cn;
    @Column(name = "org_name")
    private String on;
    private String ou;
    private String surname;
    private String givenName;
    private String o;
    private String c;
    private String e;
    //    private String s;
//    @JsonIgnore
    @Column(unique = true)
    private String subjectAlias;
    private String startDate;
    private String endDate;
    private String jksPass;

    @ManyToOne
    private Certificate issuer;
}

File: front/front-pki/src/app/model/certificate.ts
import { KeyUsage } from "../model/keyUsage"

export interface Certificate{
    issuer : string,
    serialNumber: string
    country: string,
    organizationUnit: string,
    organization : string,
    surname: string,
    commonName : string,
    email: string,
    startDate: Date|string,
    endDate: Date|string,
    keyUsage: KeyUsage,
    certificateType : number 
}